pallets/dapp-staking-v3/src/lib.rs

```diff
@@ -289,6 +296,7 @@ pub mod pallet {
             beneficiary: T::AccountId,
             smart_contract: T::SmartContract,
             tier_id: TierId,
+            rank: Rank,
```

Comment:
Hi, 
IMHO, it was better to add the new field at the end to be backward compatible with the existing clients listening your events.
Change the order of the field introduces a breaking changes for the event consumers.
Adding a new field at the end doesn't necessarily break the consumer: the new field is ignored but it still work.

Reply:
You should be able to query by key. Any reason why you are querying by index?